### PR TITLE
Dashboard: Make header names reflect the field name when exporting to CSV file from the the panel inspector

### LINKED ISSUE #24600: CSV exporter shows "Value" for all columns
**What happened**:
When exporting csv for multiple values transforming with "join by time" I can see the desired table:
Captura de tela de 2020-05-12 20-12-15
But when I export the CSV, all column names come as "Value":
```csv
Time,Value,Value,Value,Value
1589324745157.7249,0.6862745098039216,0.6666666666666666,0.6095057458440254,0.5
```

**What you expected to happen**:
The same column names displayed in preview should be exported as the first line in the CSV

**How to reproduce it (as minimally and precisely as possible)**:
- PostgreSQL datasource
- Timeseries
- Graph visualization
- X-axis as `series` and value `total`
- The (modified) query:
```sql
 SELECT
  now() as time,
  c.display_name as metric
  count(id)
FROM reviews r
INNER JOIN company c ON c.id = r.id
WHERE is_active and i.status='published'
GROUP BY 1,2
HAVING count(r.id) > 50
ORDER BY 3 DESC
```

**Anything else we need to know?**:

**Environment**: 
- Grafana version: Grafana v7.1.0-534e211cpre (5feef22034)
- Data source type & version: PostgreSQL latest
- OS Grafana is installed on: Docker
- User OS & Browser: Debian & Firefox
- Grafana plugins: None, fresh install
- Others:

### PR DESCRIPTION
**Pull request by peterholmberg**
**What this PR does / why we need it**:
Fixes the header names in the exported csv file and fixing the file name.

**Which issue(s) this PR fixes**:
Fixes #24600

**Special notes for your reviewer**:

### CODE REVIEW

File: packages/grafana-data/src/utils/csv.ts [hunk 1]
```diff
@@ -289,7 +290,7 @@ export function toCSV(data: DataFrame[], config?: CSVConfig): string {
         if (i > 0) {
           csv += config.delimiter;
         }
-        csv += fields[i].name;
+        csv += getFieldDisplayName(fields[i]);
```
Comment (hugohaggmark) [inline packages/grafana-data/src/utils/csv.ts:293]: Not sure when to pass the dataFrame props but maybe @torkelo can explain further.
```suggestion
        csv += getFieldDisplayName(fields[i], series, data);
```

**Review by hugohaggmark — COMMENTED** [review]
Great work! I think I found a minor issue but not in the naming so it might be for a separate PR.
1. new dashboard
2. enter values as picture below
image
3. export as csv 
4. the second header column contains a comma (`,`) which might mean 2 columns depending on which software parses the csv file
In Numbers for instance it looks like this:
image

File: packages/grafana-data/src/utils/csv.ts (hunk 1 above)
Comment (torkelo) [reply packages/grafana-data/src/utils/csv.ts:293]: You you should pass DataFrame here as the second argument, hmm not sure why DataFrame argument is optional :)

Comment (peterholmberg) [reply packages/grafana-data/src/utils/csv.ts:293]: but what does it do? I don't see any difference in the output?

File: packages/grafana-data/src/utils/csv.ts [hunk 2]
```diff
@@ -289,7 +290,7 @@ export function toCSV(data: DataFrame[], config?: CSVConfig): string {
         if (i > 0) {
           csv += config.delimiter;
         }
-        csv += fields[i].name;
+        csv += `"${getFieldDisplayName(fields[i], series).replace('"', '"""')}"`;
```
Comment (dprokop) [inline packages/grafana-data/src/utils/csv.ts:293]: @peterholmberg this will replace only first occurence of `"`, think you need to pass a regex

Comment (peterholmberg) [reply packages/grafana-data/src/utils/csv.ts:293]: Any pointers on that? I'm fumbling in the dark here.

Comment (dprokop) [reply packages/grafana-data/src/utils/csv.ts:293]: `replace(/"/g, '"""');`

should do the job

Comment (peterholmberg) [reply packages/grafana-data/src/utils/csv.ts:293]: I don't know, still haven't found a solution that works for LibreOffice


Screenshot from 2020-05-14 14-49-38

Comment (torkelo) [reply packages/grafana-data/src/utils/csv.ts:293]: I think this is fine for now, we cannot coverall csv parsers without an option for escape method

Comment (dprokop) [reply packages/grafana-data/src/utils/csv.ts:293]: Yeah, would be good to test the output agains Excel. MacOs Numbers has the same problem ^^

Comment (torkelo) [reply packages/grafana-data/src/utils/csv.ts:293]: Strange using double quotes to escape worked for me in numbers

Comment (torkelo) [reply packages/grafana-data/src/utils/csv.ts:293]: The replace is wrong. Try replace with only two double quotes

Comment (dprokop) [reply packages/grafana-data/src/utils/csv.ts:293]: You are right @torkelo, I haven't noticed that the change wasn't pushed

Comment (dprokop) [reply packages/grafana-data/src/utils/csv.ts]: pushed now

Comment (dprokop) [reply packages/grafana-data/src/utils/csv.ts]: Works in Excel too (O365)

**Review by dprokop — APPROVED** [review]

**Review by torkelo — APPROVED** [review]
Looks good

**Pull request merged**